server/src/test/python/cromwell-az-e2e-test/cromwell_az_e2e_test/az_e2e.py

```diff
@@ -0,0 +1,173 @@
+        'workflowInputs': ('hello.inputs', open(workflow_inputs_path, 'rb'), 'application/octet-stream'),
+        'workflowType': 'WDL',
+        'workflowTypeVersion': '1.0'
+    }
+    response = requests.post(workflow_endpoint, headers=headers, files=files)
+    handle_failed_request(response, f"Error submitting workflow to Cromwell for {workflow_test_name}", 201)
+    output_message(response.json())
+    return response.json()
+
+def get_workflow_information(app_url, workflow_id):
+    workflow_endpoint = f'{app_url}/api/workflows/v1/{workflow_id}/metadata'
+    headers = {"Authorization": f'Bearer {bearer_token}',
+              "accept": "application/json"}
+    output_message(f"Fetching workflow metadata for {workflow_id}")
+    response = requests.get(workflow_endpoint, headers=headers)
+    handle_failed_request(response, f"Error fetching workflow metadata for {workflow_id}")
+    output_message(response.json())
+    return response.json()
+
+def get_completed_workflow(app_url, workflow_ids, max_retries=4):
+    target_statuses = ['Succeeded', 'Failed']
+    current_running_workflow_count = 0
+    while workflow_ids:
+        if max_retries == 0:
+            raise Exception(f"Workflow(s) did not finish running within retry window ({max_retries} retries)")
+        workflow_id = workflow_ids.pop()
+        workflow_metadata = get_workflow_information(app_url, workflow_id)
+        if workflow_metadata['status'] in target_statuses:
+            output_message(f"{workflow_id} finished running. Status: {workflow_metadata['status']}")
+        else:
+            workflow_ids.append(workflow_id)
+            current_running_workflow_count += 1
+        if current_running_workflow_count == len(workflow_ids):
+            if current_running_workflow_count == 0:
+                output_message("Workflow(s) finished running")
+            else:
+                # Reset current count to 0 for next retry
+                # Decrement max_retries by 1
+                # Wait 2 minutes before checking workflow statuses again (adjust as needed)
+                output_message(f"These workflows have yet to return a completed status: [{', '.join(workflow_ids)}]")
+                max_retries -= 1
+                current_running_workflow_count = 0
+                time.sleep(60 * 2)
+
+def start():
+    # Giving workflow 3 minutes to complete
+    sleep_timer = 60 * 3
+
+    # Commenting out the workspace/provisioning steps for now
+    # workspace_id = create_workspace()
+    # time.sleep(60 * 20) # Added an sleep here to give the workspace time to provision
+    # app_url = get_app_url(workspace_id, 'cromwell')
+
+    # This chunk of code only executes one workflow
+    # Would like to modify this down the road to execute and store references for multiple workflows
+    workflow_response = submit_workflow_to_cromwell(app_url, "Run Workflow Test")
+    #Will need to update this when swapping out hello wdl with fetch_sra_to_bam (20 min?)
```

Comment:
What kind of update are you thinking of? Maybe sleep for a minute at a time and check for workflow status in between?

Reply:
Ah, I see that we're already doing something like that in `get_completed_workflow`. With that pause in between checks, maybe we don't need this one?